Relax --delete requirements through a new option

marciobarbosa opened the pull request:
The original cellcc start-sync --delete flag is used to delete the
specified volume in all destination cells. This option only works if
the volume in question was already removed from the source cell.
Unfortunately, this is not always desirable.

This commit introduces a new flag called --forcedelete. This option
deletes the specified volume in all destination cells even if this
volume still exists in the source cell.

adeason commented on doc/pod1/cellcc_config.pod:
```diff
@@ -154,6 +154,11 @@ This means cellcc is sending new data to the volume in the destination cell.
 
 This means cellcc is deleting the volume in the destination cell.
 
+=item C<forcedelete>
```
This doc change doesn't seem to agree with... (see below)

adeason commented on lib/AFS/CellCC.pm:
```diff
@@ -50,7 +50,7 @@ _exclude_volume($$$$$) {
 
     my $operation = 'sync';
-    if ($opts->{delete}) {
+    if ($opts->{delete} || $opts->{forcedelete}) {
```
(from above)... this code change. We still set CELLCC_FILTER_OPERATION to 'delete', whether we are deleting or force-deleting

adeason reviewed (changes requested):
The behavior of CELLCC_FILTER_OPERATION seems to disagree with the documentation, in this commit. I'm not sure which way I prefer... I'm learning towards how the code does it (no separate 'forcedelete' operation; the --forcedelete flag just causes the 'delete' operation), but I'm not sure. Either way is okay with me, but of course the documentation and code need to be in agreement.

Also, could you change the name of the command-line flag to "--force-delete"? I know --forcedelete was mentioned by the person requesting this, but I don't think they are very particular about the exact spelling, and having a separator in the words makes it a lot easier to read. (variable names and stuff don't need to change, unless you want them to)

marciobarbosa committed:
Relax --delete requirements through a new option

The original cellcc start-sync --delete flag is used to delete the
specified volume in all destination cells. This option only works if
the volume in question was already removed from the source cell.
Unfortunately, this is not always desirable.

This commit introduces a new flag called --force-delete. This option
deletes the specified volume in all destination cells even if this
volume still exists in the source cell.

marciobarbosa commented:
Branch updated.

Thank you!
